contracts/upgradeable_contracts/RewardableBridge.sol

```diff
@@ -18,38 +18,30 @@ contract RewardableBridge is Ownable, FeeTypes {
     bytes4 internal constant DISTRIBUTE_FEE_FROM_SIGNATURES = 0x59d78464; // distributeFeeFromSignatures(uint256)
     bytes4 internal constant DISTRIBUTE_FEE_FROM_AFFIRMATION = 0x054d46ec; // distributeFeeFromAffirmation(uint256)
 
-    function _getFee(bytes32 _feeType) internal view validFeeType(_feeType) returns (uint256) {
-        uint256 fee;
+    function _getFee(bytes32 _feeType) internal view validFeeType(_feeType) returns (uint256 fee) {
         address feeManager = feeManagerContract();
         bytes4 method = _feeType == HOME_FEE ? GET_HOME_FEE : GET_FOREIGN_FEE;
         bytes memory callData = abi.encodeWithSelector(method);
 
         assembly {
             let result := callcode(gas, feeManager, 0x0, add(callData, 0x20), mload(callData), 0, 32)
-            fee := mload(0)
 
-            switch result
-                case 0 {
-                    revert(0, 0)
```

Comment:
Before the changes, if the result was unsuccessful, the call reverted. It means that it could revert entire transaction. Are there cases when this code is invoked during a transaction execution rather than a method call? If so, is this code safe to be changed in this manner?

Reply:
So, here is the quick analysis. We have 5 places with such `callcode` patterns:
* `getFeeManagerMode()`, method is not called on-chain, nothing unsafe here.
* `_getFee(bytes32)` is called inside `getHomeFee()` and `getForeignFee()` of `Rewardable*Bridge*.sol` set of contracts. These methods are not called onchain inside our contracts.
* `calculateFee(uint256,bool,address,bytes32)` is called in several vanilla modes contracts.
  - In all cases, a call is located within a `if (feeManager != address(0))` condition. So, the behaviour for an unset fee manager will be exactly the same as before. 
  - For invalid fee manager contracts, zero will be returned instead of a transaction revert, meaning that setting a fee manager to an invalid contract will result in disabled fees rather than disabled bridge operations, I don't see a problem in such behaviour.
  - For valid contracts, the underlying call might only fail due to `SafeMath` exceptions. In theory, if someone want to bridge `2**200` tokens, then fee manager will revert, but the outside transaction will continue its execution assuming that fee is zero. I think it is better to revert the whole transaction in such situation, so let's ad an extra check.
* `blockRewardContract()` from `HomeBridgeErcToErcPOSDAO` is not called inside our contracts, so it is safe to return zero address.
* `getAmountToBurn(uint256)` works in exactly the same way as `calculateFee` does. We can also add an extra check there.

Reply:
Added revert statements for extra safety where needed, please take a look 10d7c0f

Reply:
thanks for analysis!